web/src/composebox_typeahead.ts

```diff
@@ -996,6 +999,57 @@ export function content_highlighter_html(item: TypeaheadSuggestion): string | un
     }
 }
 
+function will_produce_broken_stream_topic_link(word: string): boolean {
+    return invalid_stream_topic_regex.test(word);
+}
+
+function get_stream_name_from_topic_link_syntax(syntax: string): string {
+    const start = syntax.lastIndexOf("#**");
```

Comment:
Why is this `lastIndex`? This function would benefit from a comment explaining how it's intended to be used / what assumptions the caller is responsible for checking about the `syntax` before using it.

Or maybe cleaner would be to just have it start with `assert` statements for those claims.

Reply:
Both `lastIndexOf` and `indexOf` would work the same here. This is a remnant from when this piece of code wasn't separated into its own function.